File: apps/backend-api/internal/dto/gene_list.go
package dto

import (
	"time"

	"github.com/google/uuid"
	"gorm.io/datatypes"
)

// GeneListCreateRequest represents a request to create a gene list
type GeneListCreateRequest struct {
	Name            string         `json:"name" binding:"required,min=1,max=100"`
	Description     string         `json:"description" binding:"omitempty"`
	Genes           datatypes.JSON `json:"genes" binding:"required"` // Array of gene symbols
	Category        string         `json:"category" binding:"omitempty,oneof=core important optional"`
	DiseaseCategory string         `json:"diseaseCategory" binding:"omitempty,max=100"`
}

// GeneListUpdateRequest represents a request to update a gene list
type GeneListUpdateRequest struct {
	Name            *string         `json:"name" binding:"omitempty,min=1,max=100"`
	Description     *string         `json:"description" binding:"omitempty"`
	Genes           *datatypes.JSON `json:"genes" binding:"omitempty"`
	Category        *string         `json:"category" binding:"omitempty,oneof=core important optional"`
	DiseaseCategory *string         `json:"diseaseCategory" binding:"omitempty,max=100"`
}

// GeneListResponse represents a gene list response
type GeneListResponse struct {
	ID              uuid.UUID      `json:"id"`
	Name            string         `json:"name"`
	Description     string         `json:"description,omitempty"`
	Genes           datatypes.JSON `json:"genes"`
	GeneCount       int            `json:"geneCount"`
	Category        string         `json:"category,omitempty"`
	DiseaseCategory string         `json:"diseaseCategory,omitempty"`
	CreatedAt       time.Time      `json:"createdAt"`
	UpdatedAt       time.Time      `json:"updatedAt"`
}

// GeneListDetailResponse represents a detailed gene list response
type GeneListDetailResponse struct {
	ID              uuid.UUID      `json:"id"`
	Name            string         `json:"name"`
	Description     string         `json:"description,omitempty"`
	Genes           datatypes.JSON `json:"genes"`
	GeneCount       int            `json:"geneCount"`
	Category        string         `json:"category,omitempty"`
	DiseaseCategory string         `json:"diseaseCategory,omitempty"`
	CreatedAt       time.Time      `json:"createdAt"`
	UpdatedAt       time.Time      `json:"updatedAt"`
}

File: apps/web-germline/src/types/geneList.ts
// 基因列表分类
export type GeneListCategory = 'core' | 'important' | 'optional';

// 基因列表
export interface GeneList {
  id: string;
  name: string;
  description?: string;
  genes: string[];
  geneCount: number;
  category?: GeneListCategory;
  diseaseCategory?: string;
  createdAt: string;
  updatedAt: string;
}

// 创建基因列表请求
export interface GeneListCreateRequest {
  name: string;
  description?: string;
  genes: string[];
  category?: GeneListCategory;
  diseaseCategory?: string;
}

// 更新基因列表请求
export interface GeneListUpdateRequest {
  name?: string;
  description?: string;
  genes?: string[];
  category?: GeneListCategory;
  diseaseCategory?: string;
}
